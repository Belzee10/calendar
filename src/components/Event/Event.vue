<template>
  <div
    class="event text-truncate text-capitalize ma-1 px-1"
    :class="getColorClass"
    @click="emitClick"
  >
    <v-icon v-if="isPriority" x-small>
      mdi mdi-star
    </v-icon>
    {{ name }}
  </div>
</template>

<script>
export default {
  name: 'Event',
  props: {
    id: {
      type: String,
      default: ''
    },
    name: {
      type: String,
      default: ''
    },
    // TODO not sure
    date: {
      type: Object,
      default: () => ({})
    },
    type: {
      type: String,
      default: ''
    },
    isPriority: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    getColorClass() {
      return this.type === 'work' ? 'grey' : 'blue';
    }
  },
  methods: {
    emitClick() {
      this.$emit('click-on-event', {
        id: this.id
      });
    }
  }
};
</script>

<style lang="scss" scoped>
.event {
  font-size: 12px;
}
</style>
