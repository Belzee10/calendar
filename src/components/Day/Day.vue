<template>
  <v-sheet>
    <Event v-for="event in events" :key="event.id" v-bind="event" />
  </v-sheet>
</template>

<script>
import Event from '@/components/Event/Event.vue';
export default {
  name: 'Day',
  components: { Event },
  props: {
    events: {
      type: Array,
      default: () => []
    },
    day: {
      type: String,
      default: ''
    }
  }
};
</script>

<style lang="scss" scoped></style>
