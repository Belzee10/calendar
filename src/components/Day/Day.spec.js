import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Day from './Day.vue';
import Event from '@/components/Event/Event.vue';

const localVue = createLocalVue();

describe('Day.vue', () => {
  let vuetify;
  beforeEach(() => {
    vuetify = new Vuetify();
  });
  const props = {
    events: [
      {
        id: 'ck1zaeuof0000l92p1kwi1g7i',
        name: 'quod enim et',
        start: '2019-10-04'
      },
      {
        id: 'ck1zaeuog0001l92p58yg4qk7',
        name: 'voluptatem aperiam recusandae',
        start: '2019-10-04'
      }
    ]
  };
  test('should render the "events" by day', () => {
    const wrapper = mount(Day, {
      vuetify,
      localVue,
      propsData: props
    });
    expect(wrapper.findAll(Event)).toHaveLength(2);
  });

  test('should render correclty', () => {
    const wrapper = mount(Day, {
      vuetify,
      localVue,
      propsData: props
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
