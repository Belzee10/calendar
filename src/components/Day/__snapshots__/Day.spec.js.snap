// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day.vue should render correclty 1`] = `
<div
  class="v-sheet theme--light"
>
  <div
    class="event text-truncate text-capitalize ma-1 px-1 blue"
    start="2019-10-04"
  >
    <!---->
    
  quod enim et

  </div>
  <div
    class="event text-truncate text-capitalize ma-1 px-1 blue"
    start="2019-10-04"
  >
    <!---->
    
  voluptatem aperiam recusandae

  </div>
</div>
`;
